platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/schedulers/builders/TaskSchedulerConfiguration.java

```diff
@@ -0,0 +1,141 @@
+/**
+ * Configures a task scheduler.
+ *
+ * @param type                       the type of task scheduler, if null then {@link TaskSchedulerType#SEQUENTIAL} is
+ *                                   used
+ * @param unhandledTaskCapacity      the maximum number of unhandled tasks, or 0 if unbounded, if null then 0 is used
+ * @param unhandledTaskMetricEnabled whether the unhandled task count metric should be enabled, if null than false is
+ *                                   used
+ * @param busyFractionMetricEnabled  whether the busy fraction metric should be enabled, if null then false is used
+ * @param flushingEnabled            whether flushing is enabled, if null then false is used
+ * @param squelchingEnabled          whether squelching is enabled, if null then false is used
+ */
+public record TaskSchedulerConfiguration(
+        @Nullable TaskSchedulerType type,
+        @Nullable Long unhandledTaskCapacity,
+        @Nullable Boolean unhandledTaskMetricEnabled,
+        @Nullable Boolean busyFractionMetricEnabled,
+        @Nullable Boolean flushingEnabled,
+        @Nullable Boolean squelchingEnabled) {
+
+    /**
+     * Parse a string representation of a task scheduler configuration.
+     * <p>
+     * Syntax is as follows:
+     * <ul>
+     *     <li>
+     *         Zero or one values from the {@link TaskSchedulerType} enum, specifies the type of the task scheduler.
+     *         If not present then the default is used.
+     *     </li>
+     *     <li>
+     *         Zero or one string of the form "CAPACITY(1234)", specifies the maximum number of unhandled tasks.
+     *     </li>
+     *     <li>
+     *         Zero or more values from the {@link TaskSchedulerConfigOption} enum, specifies the configuration options.
+     *         Sets a boolean configuration option to true if the value is present, and false if the value is prefixed
+     *         with a "!". If not present then the default is used.
+     *     </li>
+     * </ul>
+     * Example: "SEQUENTIAL CAPACITY(500) !FLUSHABLE UNHANDLED_TASK_METRIC"
+     * <p>
+     * Note that default values are not specified within this class. Default values are the responsibility of the
+     * {@link TaskSchedulerBuilder} class.
+     *
+     * @param string the string to parse
+     * @return the parsed configuration
+     */
+    @NonNull
+    public static TaskSchedulerConfiguration parse(@NonNull final String string) {
+        TaskSchedulerType type = null;
+        Long unhandledTaskCapacity = null;
+        Boolean unhandledTaskMetricEnabled = null;
+        Boolean busyFractionMetricEnabled = null;
+        Boolean flushingEnabled = null;
+        Boolean squelchingEnabled = null;
+
+        try {
+            final String[] parts = string.split(" ");
+            for (final String part : parts) {
+                String strippedPart = part.strip();
+                if (strippedPart.isEmpty()) {
+                    continue;
+                }
+
+                try {
```

Comment:
consider splitting this `parse` method into separate private methods: `parseType`, `parseCapacity`, and `parseOption`

Reply:
I've split a large amount of the logic into helper functions, but due to the nature of variables being local to the static method and not classmember variables, there is still a bunch of logic that has to remain in the top level method.